dex/src/state.rs

```diff
@@ -2686,8 +2691,201 @@ impl State {
+            coin_wallet,
+            pc_wallet,
+            coin_vault,
+            pc_vault,
+            spl_token_program,
+            fee_tier,
+            vault_signer
+        } = args;
+
+
+        let limit_price = instruction.limit_price;
+        let mut limit = instruction.limit;
+        let min_coin_qty = instruction.min_coin_qty;
+        let min_native_pc_qty = instruction.min_native_pc_qty;
+
+        let order_type = OrderType::ImmediateOrCancel;
+        let self_trade_behavior = SelfTradeBehavior::AbortTransaction;
+
+        let order_id = req_q.gen_order_id(instruction.limit_price.get(), instruction.side); 
+        let mut proceeds = RequestProceeds::zero();
+        
+        let native_pc_qty_locked = match instruction.side {
+            Side::Bid => {
+                let lock_qty_native = instruction.max_native_pc_qty_including_fees;
+                Some(lock_qty_native)
+            }
+            Side::Ask => {
+                None
+            }
+        };
+
+        let request = RequestView::NewOrder {
+            side: instruction.side,
+            order_type,
+            order_id,
+            fee_tier,
+            self_trade_behavior,
+            // Pass dummy account - system program pubkey
+            owner: system_program::ID.to_aligned_bytes(),
+            // Pass 0 
+            owner_slot: 0,
+            max_coin_qty: instruction.max_coin_qty,
+            native_pc_qty_locked,
+            client_order_id: None,
+        };
+
+        let _unfilled_portion = order_book_state.process_orderbook_request(
+            &request,
+            &mut event_q,
+            &mut proceeds,
+            &mut limit,
+        )?;
+
+        let mut market_state = order_book_state.market_state;
+
+        let pc_lot_size = market_state.pc_lot_size;
+        let coin_lot_size = market_state.coin_lot_size;
+
+        let RequestProceeds {
+            coin_unlocked,
+            coin_credit,
+
+            native_pc_unlocked, 
+            native_pc_credit,
+
+            coin_debit,
+            native_pc_debit,
+        } = proceeds;
+
+        let mut abort= false;
+
+        match instruction.side {
+            Side::Bid => {
+                if coin_credit < min_coin_qty {
+                    abort = true;
+                };
+            },
+            Side::Ask => {
+                if native_pc_credit < min_native_pc_qty {
+                    abort = true;
+                }
+            }
+        }
+
+        if abort {
+            solana_program::msg!("Min amount requested not met! Aborting");
+            return Err(DexErrorCode::MinAmountNotMet.into())
+        };
+
+        let native_debit;
+        let native_credit;
+        let deposit_vault;
+        let payer;
+        let recipient;
+        let payer_vault;
+        let native_taker_fee;
+
+        match instruction.side {
+            Side::Bid => {
+                native_debit = native_pc_debit;
```

Comment:
Why don't we multiply by `pc_lot_size` here? Is the multiplication needed or not?

Reply:
Because native_pc_debit is already native. Aka, on the client side, we pass in pc_quantity * pc_lot_size. Therefore, we don't need to multiply by pc_lot_size in most of process_new_order_v3 or process_send_take because our inputs are native